public_html/wp-content/mu-plugins/blocks/assets/src/shared/featured-image/index.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies.
+ */
+const { Component, Fragment, RawHTML } = wp.element;
+
+/**
+ * Displays featured image, can be linked with block control for size. Fetches the smallest possible image size required to render the current size.
+ */
+export default class FeaturedImage extends Component {
+
+	/**
+	 * Constructor for component. Size can be provided, if height and width both are provided, then they take precedence over size.
+	 *
+	 * @param props Props for function.
+	 * @oaram props.wpMediaDetails Available sizes of images in the format as returned by WP API. This is the `sizes` object inside `media_details` inside `wp:featuredMedia` object. Not required if `props.availableSizes is present.
+	 * @param props.availableSizes Available sizes of image along with source URL. Not required if `props.wpMediaDetails` is present. Should be in this format:
+	 * [
+	 * 		{
+	 * 		 	height: 999,
+	 * 		 	width: 999,
+	 * 		 	source_url: https://link.to/image-size-1.png
+	 * 		},
+	 * 		{
+	 * 			height: 500,
+	 * 			width: 500,
+	 * 			source_url: https://link.to/image-size-2.png
+	 * 		}
+	 * 		...
+	 * ]
+	 * @param props.height Height in pixels for image.
+	 * @param props.width Width in pixels for image.
+	 * @param props.className Classname for image element
+	 * @param props.alt Alt text for image
+	 */
+	constructor( props ) {
+		super( props );
+	}
+
+	componentWillMount() {
+		let availableSizes;
+		const { wpMediaDetails } = this.props;
+		if ( wpMediaDetails ) {
+			availableSizes = FeaturedImage.parseWPMediaDetails( wpMediaDetails );
+		} else {
+			availableSizes = this.props.availableSizes;
+		}
+		this.parseAvailableSizes( availableSizes );
+	}
+
+	/**
+	 * Converts from WP API featured_media object to array of different sizes.
+	 *
+	 * @param wpMediaDetails
+	 */
+	static parseWPMediaDetails( wpMediaDetails ) {
+		const availableSizes = [];
+		for ( const size of Object.keys( wpMediaDetails ) ) {
+			if ( ! wpMediaDetails.hasOwnProperty( size  ) ) {
+				continue;
+			}
+
+			availableSizes.push( wpMediaDetails[ size ] );
+		}
+		return availableSizes;
+	}
+
+	/**
+	 * Pre-calculate and store aspect ratio of all images. This will help us in picking the best image available for the size that we have.
+	 */
+	parseAvailableSizes( availableSizes ) {
+		let largestSizeValue = 0;
+		let largestSize = {};
```

Comment:
It looks like these two variables aren't used...

Reply:
fixed